target/linux/ramips/dts/DWR-512-B.dts

```diff
@@ -0,0 +1,115 @@
+/dts-v1/;
+
+/include/ "rt5350.dtsi"
+
+/ {
+	compatible = "ralink,rt5350-soc";
+	model = "D-Link DWR-512 B";
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 0 1>;
+			linux,code = <0x211>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		sms {
+			label = "dwr-512-b:green:sms";
+			gpios = <&gpio0 8 1>;
+		};
+		status {
+			label = "dwr-512-b:green:status";
+			gpios = <&gpio0 9 1>;
+		};
+		2g {
+			label = "dwr-512-b:green:2g";
+			gpios = <&gpio0 17 1>;
+		};
+		3g {
+			label = "dwr-512-b:green:3g";
+			gpios = <&gpio0 19 1>;
+		};
+		sstrengthr {
+			label = "dwr-512-b:red:sigstrength";
+			gpios = <&gpio0 20 1>;
+		};
+		sstrengthg {
+			label = "dwr-512-b:green:sigstrength";
+			gpios = <&gpio0 21 1>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+ 
+		modem3g_enable {
+			gpio-export,name = "modem3g_enable";
+			gpio-export,output = <1>;
+			gpios = <&gpio0 18 1>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	mx25l6405d@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "macronix,mx25l6405d", "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <30000000>;
+		fast-read;
+
+                partition@0 {
+                        label = "u-boot";
+                        reg = <0x0 0x30000>;
+                        read-only;
+                };
+
+                partition@30000 {
+                        label = "u-boot-env";
+                        reg = <0x30000 0x10000>;
+                        read-only;
+                };
+
+                factory: partition@40000 {
+                        label = "factory";
+                        reg = <0x40000 0x10000>;
+                        read-only;
+                };
+
+                partition@50000 {
+                        label = "firmware";
+                        reg = <0x50000 0x7a0000>;
+                };
+
+		config: partition@7f0000 {
+			label = "config";
+			reg = <0x7f0000 0x10000>;
+		};
+	};
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "jtag", "uartf";
+			ralink,function = "gpio";
+		};
+	};
+};
+
+&wmac {
+	ralink,mtd-eeprom = <&config 0xe07a>;
```

Comment:
here it is:
root@lede:~# hexdump -C /dev/mtd2 | head
00000000  33 6e 70 e0 b9 95 cc 56  34 6b 21 87 3f c4 3f 06  |3np....V4k!.?.?.|
00000010  34 b9 0e 24 66 d6 50 c0  ca b4 a1 05 e1 df 76 37  |4..$f.P.......v7|
00000020  96 bb 4a 0f c5 d4 7b 0e  d7 cf 03 49 39 ed 1e d2  |..J...{....I9...|
00000030  0a bd 6d 7c f7 2e 55 5a  f4 47 7c eb 06 d6 b7 24  |..m|..UZ.G|....$|
00000040  74 84 54 3d 41 0b 90 1d  ef 09 b4 f5 ef 8e 81 ab  |t.T=A...........|
00000050  a1 43 ea 78 89 ff a8 48  45 bd b5 bf 8d b9 44 8f  |.C.x...HE.....D.|
00000060  e7 c2 dc a6 83 91 dc 80  e6 ef 6e 07 c2 05 26 33  |..........n...&3|
00000070  60 8f 66 3e d6 e4 e7 e7  37 2b 58 45 ca 5b d4 de  |`.f>....7+XE.[..|
00000080  2c ea 77 0b 67 f5 b3 19  ab 88 43 0c e7 dc cc 5f  |,.w.g.....C...._|
00000090  9f 77 be 0a 35 19 f8 7b  28 f2 78 ea 50 c9 c1 91  |.w..5..{(.x.P...|